static/app/utils/replays/Replay.tsx

```diff
@@ -0,0 +1,96 @@
+import memoize from 'lodash/memoize';
+import type {eventWithTime} from 'rrweb/typings/types';
+
+import type {RawSpanType} from 'sentry/components/events/interfaces/spans/types';
+import type {RawCrumb} from 'sentry/types/breadcrumbs';
+import type {Event, EventTransaction} from 'sentry/types/event';
+import {EntryType} from 'sentry/types/event';
+import mergeBreadcrumbEntries from 'sentry/utils/replays/mergeBreadcrumbEntries';
+import mergeSpanEntries from 'sentry/utils/replays/mergeSpanEntries';
+
+function last<T>(arr: T[]): T {
+  return arr[arr.length - 1];
+}
+
+export default class Replay {
+  static factory(
+    event: EventTransaction | undefined,
+    rrwebEvents: eventWithTime[] | undefined,
```

Comment:
Thinking we should re-name this so that it's not tied to rrweb, but for another time.